### PR TITLE
Merge index tests

### PR DESCRIPTION
**Pull request by ibilal**
Merged all index GET/POST tests into a single file.

### CODE REVIEW

File: test/routes/index.js [hunk 1]
```diff
@@ -54,7 +60,33 @@ describe('in index/routes ', function () {
...
     }
   ];
-  describe('router.post function', function () {
+
+  describe('get request', function () {
+    it('should contain a property called text', function (done) {
+      // This makes a server request to the route location '/'
+      chai.request(server)
+            .get('/')
+            .end(function (err, res, body) {
```
Comment (ibilal) [inline test/routes/index.js]: There is a parameter body here, but it's never used. Do we need it?

Comment (chances) [inline test/routes/index.js]: Nope, you're right; if it does nothing then we don't need it.

File: test/routes/index.js [hunk 2]
```diff
@@ -1,17 +1,23 @@
 /* eslint-env mocha */
+
 let chai = require('chai');
 let expect = chai.expect;
+let should = chai.should;
```
Comment (chances) [inline test/routes/index.js]: Remove `should`.

**Comment by chances** [comment]
Apart from unused should, LGTM.

File: test/routes/index.js [hunk 3]
```diff
@@ -54,7 +60,33 @@ describe('in index/routes ', function () {
...
-  describe('router.post function', function () {
+
+  describe('get request', function () {
+    it('should contain a property called text', function (done) {
+      // This makes a server request to the route location '/'
+      chai.request(server)
+            .get('/')
+            .end(function (err, res, body) {
+              if (err) {
+                should.fail('Index Page Failed.');
```
Comment (chances) [inline test/routes/index.js]: Replace with expect flavor.

**Comment by chances** [comment]
Apart from rogue should, LGTM.

File: test/routes/index.js (hunk 2 above)
Comment (ibilal) [inline test/routes/index.js]: lint complained about removing should

File: test/routes/index.js (hunk 3 above)
Comment (ibilal) [inline test/routes/index.js]: can I say expect.fail? How do I know what the err would be equal to, to maybe say expect(err) to equal ('blah')

File: test/routes/index.js (hunk 2 above)
Comment (ibilal) [inline test/routes/index.js]: nvm this was resolved when I changed the next note to expect

File: test/routes/index.js [hunk 4]
```diff
@@ -81,25 +113,26 @@ describe('in index/routes ', function () {
...
+
       chai.request(index)
-        .post('/')
-        .field('username', fixture.username)
-        .field('password', fixture.password)
-        .field('teamName', fixture.teamName);
+              .post('/')
+              .field('username', fixture.username)
+              .field('password', fixture.password)
+              .field('teamName', fixture.teamName);
```
Comment (kathtran) [inline test/routes/index.js]: Did the linter accept all of the extraneous whitespace?

Comment (chances) [inline test/routes/index.js]: All the checks passed

Comment (chances) [inline test/routes/index.js]: Maybe try linting the branch locally?

Comment (kathtran) [inline test/routes/index.js]: So it's fine to merge?

Comment (chances) [inline test/routes/index.js]: GitHub is still showing the rogue should. So no, that's unresolved

Comment (ibilal) [inline test/routes/index.js]: @chances I wasn't sure if expect.fail() is correct. It worked, but it wasn't in the docs I saw it on a site.

Comment (chances) [inline test/routes/index.js]: @ibilal https://github.com/chaijs/chai/issues/487

**Comment by ibilal** [comment]
Before you merge this branch, I noticed that I get this printed to the console and I'm not sure if it's supposed to be doing this: 
double callback!
double callback!
double callback!
double callback!
double callback!
double callback!
double callback!
@chances @MichaelLHansen @ruv-prog-so is this fine? This happens after running all mocha tests.

**Comment by chances** [comment]
@ibilal Another feature branch was merged with leftover console log's in the index get request test.

**Comment by RickyV33** [comment]
@chances doesn't look like there are any console logs in that test. Might be a warning from node about an actual double callback?

**Pull request merged**